alphagov/govuk-coronavirus-vulnerable-people-form
# Codecov GitHub action

What
----

- Codecov is a code coverage reporting tool. It checks that your tests still test the same (or more) of your code when you add new code. This is an example report from the docs: https://codecov.io/gh/codecov/example-ruby.

- Codecov is free for public repos.

- This will enable us to identify any gaps in our test coverage.

- There is a Github Marketplace app (https://github.com/marketplace/codecov) for Codecov which has already been configured for all public AlphaGov repositories.

- There is also a Github Marketplace Actions app (https://github.com/marketplace/actions/codecov) for Codecov.

- Reports need to be in XML format to be uploaded to Codecov, so there is a commit to generate the coverage report as XML.

- Uploading the report to Codecov as a Github Action is configured within this PR.

How to review
-------------

:ship: Since this service is continuously deployed, all testing must be done
before the pull request is merged. :ship:

- Have a look at https://github.com/marketplace/actions/codecov and see if you think this looks ok in comparison. 

- Its hard to test if it has worked until it runs... once it has then:

1. Go to: https://codecov.io 
2. Sign in with Github
3. Choose the govuk-coronavirus-vulnerable-people-form repo from the menu
4. You should see some metrics
5. A bot also posts comments in Github

Links
-----

[Trello](https://trello.com/c/iKWfHb9n/437-add-codecov-to-our-ci-pipelines)
